internal/services/compute/capacity_reservation_resource.go

```diff
@@ -0,0 +1,282 @@
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceCapacityReservationCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationsClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	capacityReservationGroupName := d.Get("capacity_reservation_group_name").(string)
+	id := parse.NewCapacityReservationID(subscriptionId, resourceGroup, capacityReservationGroupName, name)
+	existing, err := client.Get(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, "")
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
+		}
+	}
+	if !utils.ResponseWasNotFound(existing.Response) {
+		return tf.ImportAsExistsError("azurerm_capacity_reservation", id.ID())
+	}
+
+	parameters := compute.CapacityReservation{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Sku:      expandCapacityReservationSku(d.Get("sku").([]interface{})),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("zone"); ok {
+		zones := []string{v.(string)}
+		parameters.Zones = &zones
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, parameters)
+	if err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceCapacityReservationRead(d, meta)
+}
+
+func resourceCapacityReservationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Capacity Reservation %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("capacity_reservation_group_name", id.CapacityReservationGroupName)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("sku", flattenCapacityReservationSku(resp.Sku)); err != nil {
+		return fmt.Errorf("setting `sku`: %+v", err)
+	}
+
+	if zones := resp.Zones; zones != nil && len(*zones) > 0 {
+		d.Set("zone", (*zones)[0])
+	}
```

Comment:
The API exposed `zones`, however, it can only accept one `zone`, so used the non-common way to expand/flatten this. Similar to `managed_disk.zone` https://github.com/hashicorp/terraform-provider-azurerm/blob/7c27c458db592eb8ffc47b2fe646f0b00f399f48/internal/services/compute/managed_disk_resource.go#L472-L476